polysh/buffered_dispatcher.py

```diff
@@ -55,10 +55,17 @@ def handle_read(self):
                         break
                     else:
                         raise
-                new_data += piece
-                buffer_length += len(piece)
+
+                if piece:
+                    new_data += piece
+                    buffer_length += len(piece)
+                else:
+                    # A closed connection is indicated by signaling a read
+                    # condition, and having recv() return 0.
+                    break
```

Comment:
Just adding `if not piece: break` would be cleaner.

Reply:
A procedural question here: After a review, would you like me to squash/fixup these changes in the original commit or create a new commit fixing this?

Reply:
I think most would be better squashed as Python 3 migration.  We would never want to see the intermediate non-working versions in the future.  Of course, it would be nicer if we can keep changes other than Python 3 migration as separate commits.

Reply:
I fixed everything but this one. Do you really find
```
if piece:
    [...]

if not piece:
    [...]
```
more pretty then the way it is done right now?

Reply:
If you check for `not piece` before and `break`, then the main part doesn't need to be indented.

Reply:
Ah, I get it now. That's indeed more pretty.